### ISSUE TITLE
Route handler registered twice

### ISSUE THREAD

**Issue by Saniol** [fastify/fastify-rate-limit#120]
## 🐛 Bug Report

When using fastify with `ignoreTrailingSlash: false` (default), creating routes with prefixing and defining rate limitting for `/` of prefixed route, each request is counted twice. As I looked deeper into this issue it looks like `onRequest` handler is added twice - first time when fastify adds route for '' (node_modules\fastify\lib\route.js:163), and second time when it adds route for '/' (node_modules\fastify\lib\route.js:166; fastify-rate-limit onRequest handler seems to already exist in routeOptions then).

## To Reproduce

```js
  const subroute = async (fst) => {
      fst.register(FastifyRateLimit, {
        max: 60,
        timeWindow: 1000,
        keyGenerator: (req) => {
          console.log('call'); // will be called twice when POST /resource called

          return 'test';
        },
      });

      fst.post('/', {}, (req, res) => {
        res.type('application/json').send({ data: 'test' });
      });
    };

    const app = fastify();

    app.register(subroute, { prefix: '/resource' });
```

## Expected behavior

`onRequest` called once and each request counted as one.

## Your Environment

- *node version*: 12
- *fastify version*: 3.9.2
- *os*: Windows

**Comment by Eomm** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

We closed an issue similar to this one here https://github.com/fastify/fastify-compress/pull/141 if you would like to take inspiration

**Comment by Saniol** [comment]
@mcollina @salmanm guys I see that my fix was removed, if you'll check tests @5.3.0 the one checking that issue is failing now
